drivers/led_strip/apa102.c

```diff
@@ -21,7 +21,17 @@ static int apa102_update(const struct device *dev, void *buf, size_t size)
 {
 	const struct apa102_config *config = dev->config;
 	static const uint8_t zeros[] = { 0, 0, 0, 0 };
-	static const uint8_t ones[] = { 0xFF, 0xFF, 0xFF, 0xFF };
+
+	/*
+	 * the only function of the “End frame” is to supply more clock pulses
+	 * to the string until the data has permeated to the last LED. The
+	 * number of clock pulses required is exactly half the total number
+	 * of LEDs in the string
+	 */
+	uint8_t ones[((size / sizeof(struct led_rgb)) / 2) + 1];
```

Comment:
Wondering if it is really a good idea to allocate a dynamically sized array on the stack, especially since it's just filled with 0xff, this could waste an awful lot of RAM on a long strip.

How about allocating an 0xff array statically up to a certain configurable size? The drawback is then that the maximum strip length would have to be configured at build time but imo that's way better than changing this as potentially blowing up the stack

Reply:
Hi @fabiobaltieri. Thanks for spotting this.

@fouge Please can you statically `initialize` the ones array (with the half the strip length) and pass it via the `dev->data` pointer ?

Reply:
Can't say I haven't done it myself. :-)